control-base/src/main/java/fi/nls/oskari/control/data/MapLayerGroupsHandler.java

```diff
@@ -0,0 +1,145 @@
+        ResponseHelper.writeResponse(params, result);
+    }
+
+    /**
+     * Handles insert
+     * @param params
+     * @throws ActionException
+     */
+    public void handlePut(ActionParameters params) throws ActionException {
+        params.requireAdminUser();
+        MaplayerGroup maplayerGroup = populateFromRequest(params.getPayLoadJSON());
+        final int id = oskariMapLayerGroupService.insert(maplayerGroup);
+        // check insert by loading from DB
+        final MaplayerGroup savedMapLayerGroup = oskariMapLayerGroupService.find(id);
+        ResponseHelper.writeResponse(params, savedMapLayerGroup.getAsJSON());
+    }
+
+    /**
+     * Handles update
+     * @param params
+     * @throws ActionException
+     */
+    public void handlePost(ActionParameters params) throws ActionException {
+        params.requireAdminUser();
+        MaplayerGroup maplayerGroup = populateFromRequest(params.getPayLoadJSON());
+        maplayerGroup.setId(params.getRequiredParamInt(PARAM_ID));
+        oskariMapLayerGroupService.update(maplayerGroup);
+        ResponseHelper.writeResponse(params, maplayerGroup.getAsJSON());
+    }
+
+    /**
+     * Handles removal
+     * @param params
+     * @throws ActionException
+     */
+    public void handleDelete(ActionParameters params) throws ActionException {
+        params.requireAdminUser();
+        final int id = params.getRequiredParamInt(PARAM_ID);
+        final MaplayerGroup theme = oskariMapLayerGroupService.find(id);
+        final List<Integer> maplayerIds = oskariMapLayerGroupService.findMaplayersByGroup(id);
+        if(!maplayerIds.isEmpty()) {
+            // theme with maplayers under it can't be removed
+            throw new ActionParamsException("Maplayers linked to theme", JSONHelper.createJSONObject("code", "not_empty"));
+        }
+        oskariMapLayerGroupService.delete(id);
+        ResponseHelper.writeResponse(params, theme.getAsJSON());
+    }
+
+    private MaplayerGroup populateFromRequest(JSONObject mapLayerGroupJSON) throws ActionException {
+        MaplayerGroup maplayerGroup = new MaplayerGroup();
+        try{
+            JSONObject locales = mapLayerGroupJSON.getJSONObject(KEY_LOCALES);
+            int parentId = mapLayerGroupJSON.optInt(KEY_PARENT_ID, -1);
+            boolean selectable = mapLayerGroupJSON.optBoolean(KEY_SELECTABLE, true);
+            Iterator<?> keys = locales.keys();
+
+            while( keys.hasNext() ) {
+                String locale = (String)keys.next();
+                String name = locales.getString(locale);
+                maplayerGroup.setName(locale, name);
+            }
+            maplayerGroup.setParentId(parentId);
+            maplayerGroup.setSelectable(selectable);
+        } catch(JSONException ex) {
+            log.error("Cannot populate maplayergroup from request", ex);
+            throw new ActionException("Cannot populate maplayergroup from request");
```

Comment:
Don't log and throw as it leads to excessive logging. So you can remove the log.error here and add the "cause exception" to ActionException.